src/db/dbconfig.ts

```diff
@@ -1,17 +1,31 @@
 import { DATABASE_URL } from '@core';
 import { User } from '@entities';
+import { isTest } from 'test';
 import { Container } from 'typedi';
 import { DataSource } from 'typeorm';
 
-export const dataORM = new DataSource({
+const dataORMOpts: DataSource = new DataSource({
   type: 'postgres',
   entities: [User],
   synchronize: true,
 });
 
-export async function dbConfig() {
-  dataORM.setOptions({
-    url: Container.get(DATABASE_URL),
-  });
-  await dataORM.initialize();
+export class Database {
+  static readonly dataORM: DataSource = dataORMOpts;
+  protected db: DataSource;
+  private url: string;
+
+  constructor() {
+    this.db = Database.dataORM;
+    this.url = Container.get(DATABASE_URL);
+  }
+
+  async init(): Promise<void> {
+    this.db.setOptions({
+      connectionTimeout: isTest ? 2000 : undefined,
+      applicationName: 'NodeJS',
+      url: this.url,
+    });
+    await this.db.initialize();
+  }
```

Comment:
Vc nao precisa desse constructor e da pra mudar o protected para private:

```suggestion
export class Database {
  static readonly dataORM: DataSource = dataORMOpts;
  private readonly db: DataSource = dataORMOpts;
  private readonly url = Container.get(DATABASE_URL);

  async init(): Promise<void> {
    this.db.setOptions({
      connectionTimeout: isTest ? 2000 : undefined,
      applicationName: 'NodeJS',
      url: this.url,
    });
    await this.db.initialize();
  }
```